pkg/arvo/ted/eth-provider.hoon

```diff
@@ -0,0 +1,100 @@
+/-  spider, eth-provider, json-rpc
+/+  strandio, ethio, ethereum
+=,  strand=strand:spider
+=,  dejs-soft:format
+=,  strand-fail=strand-fail:libstrand:spider
+=,  jael
+=<
+^-  thread:spider
+|=  arg=vase
+=/  m  (strand ,vase)
+^-  form:m
+=/  eth-input  !<(ethin:eth-provider arg)
+;<  =bowl:spider  bind:m  get-bowl:strandio
+?>  =(src.bowl our.bowl)
+;<    state=state:eth-provider
+    bind:m
+  (scry:strandio state:eth-provider /gx/eth-provider/get-state/noun)
+?-  active.state
+  %local
+(call-ethio eth-input active.state url.local.state)
+  %provider
+(call-ethio eth-input active.state url.provider.state)
+  %client
+;<  ~  bind:m
+  (watch:strandio /updates [client.state %eth-provider] [%updates tid.bowl ~])
+;<  ~  bind:m
+  %+  poke:strandio 
+    [provider.client.state %eth-provider] 
+  [%provider-action !>([%provide tid.bowl eth-input])]
+;<  =cage  bind:m  (take-fact:strandio /updates)
+(pure:m q.cage)
+==
+::
+|%
++$  provider-state
+  $:  active=active:eth-provider
+      local=local:eth-provider
+      provider=provider:eth-provider
+      client=client:eth-provider
+  ==
+++  is-client
+  |=  [tid=@tatid active=active:eth-provider eth-output=ethout:eth-provider]
+  =/  m  (strand ,vase)
+  ?-  active
+    %local
+  (pure:m !>(eth-output))
+    %provider
+  (pure:m !>([tid eth-output]))
+    %client
+  (pure:m !>(eth-output))
+  ==
+++  call-ethio
```

Comment:
I think this (and maybe some of the other logic here) is already implemented in `/lib/eth-provider`, so can de deduplicated here.